### PR TITLE
[crypto] make sure MAC keys are exportable

### PR DESCRIPTION
**Pull request by edmont**
If MAC keys are not exportable their reference can't be used by the
platform's radio implementation to retrieve the actual keys.

### CODE REVIEW

**Comment by hemanth-silabs** [comment]
@edmont Any particular reason to have these exportable? It was intentionally made non-exportable, because I did not see any specific reason to export the MAC keys.

**Comment by edmont** [comment]
> @edmont Any particular reason to have these exportable? It was intentionally made non-exportable, because I did not see any specific reason to export the MAC keys.

For the case of a radio which supports handling of the frames security (`OT_RADIO_CAPS_TRANSMIT_SEC`) but without PSA capabilities. However, I understand your concerns about making the keys exportable by default, so I'll modify the PR to make this option configurable.

File: src/core/config/platform.h [hunk 1]
```diff
@@ -148,6 +148,16 @@
...
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_PLATFORM_KEYS_EXPORTABLE_ENABLE
+ *
+ * Define to 1 if you want to make MAC keys exportable.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_PLATFORM_KEYS_EXPORTABLE_ENABLE
+#define OPENTHREAD_CONFIG_PLATFORM_KEYS_EXPORTABLE_ENABLE 0
```
Comment (hemanth-silabs) [inline src/core/config/platform.h:158]: NitPick: Maybe we can rename the macro to `OPENTHREAD_CONFIG_PLATFORM_MAC_KEYS_EXPORTABLE_ENABLE`

**Review by hemanth-silabs — APPROVED** [review]

File: src/core/config/platform.h (hunk 1 above)
Comment (hemanth-silabs) [reply src/core/config/platform.h:158]: @edmont LGTM 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**